### PR TITLE
Use explicit points for following and hidden tags

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
The count in the navigation item for Following Tags and Hidden Tags was not always corresponding to the actual number of hidden and following tags.

This was because `followed_by` and `antifollowed_by` were using `points` instead of `explicit_points`. 

Note this bug affects the /tags page as well, but I have not yet fixed it there because it uses `currentUser.followed_tags`

## Related Tickets & Documents

- Related Issue #https://github.com/forem/forem/issues/19998
- Closes #https://github.com/forem/forem/issues/19998

## QA Instructions, Screenshots, Recordings
Explanation of the bug can be found in the recording. 

This cannot really be tested on Uffizi because we no longer can set weights on the following tags pages, they default to 1. Hence its better to test it locally. 

1. Find a tag that is showing on the dashboard/following_tags. You can use something like: 
```
 f = Follow.where(explicit_points: 1, followable_type: "ActsAsTaggableOn::Tag", follower_id: <your logged in user id>).last
```
The record you get back will look like this:
```
=> #<Follow:0x000000010ddc9de0
 id: 26,
 followable_type: "ActsAsTaggableOn::Tag",
 followable_id: 37,
 follower_type: "User",
 follower_id: 73,
 blocked: false,
 created_at: Tue, 01 Aug 2023 18:01:22.569443000 UTC +00:00,
 updated_at: Tue, 01 Aug 2023 18:01:22.569443000 UTC +00:00,
 points: 1.0,
 subscription_status: "all_articles",
 explicit_points: 1.0,
 implicit_points: 0.0>
```
2. Make sure that you know which tag this is for so that you can observe it on the UI. YOu can determine the tag by running:
```
tag = Tag.find(f.followable_id)
```
3. Now update the `implicit_points` to -2 of the follow so that we turn `points` from positive to negative. Note: `points` is generated in the background based on `implicit_points` and `explicit_points`.
```
f.implicit_points = -2
f.save!
```
4. Check that the points have changed from a positive to a negative number: 
```
f.points <should be a negative number>
```
```
f.explicit_points <should still be a 1>
```

5. On `main` you should see a mismatch on the navigation count vs the actual page. On this branch the count should be aligned because we're filtering by `explicit_points` and not `points`.

Sorry for the longer video:
https://www.loom.com/share/200e356b97f4428488b3b299ad813ced?sid=db26ec6b-d0b6-4243-afcb-d349426bb0d0

### UI accessibility checklist

N/A

## Added/updated tests?

- [x] Yes, kind of, I added implicit_points to make sure that we're taking into account only explicit points.
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: spec/models/tag_spec.rb [hunk 1]
```diff
@@ -229,6 +229,7 @@
 
       antifollowed = user.follow(antifollowed_tag)
       antifollowed.update explicit_points: -5
+      antifollowed.update implicit_points: 6
```
Comment (lightalloy) [inline spec/models/tag_spec.rb:232]: Could `update(explicit_points: -5, implicit_points: 6)` at once, so optimize a bit?

**Review by lightalloy — APPROVED** [review]
I have reproduced the bug using instructions in the description + tried the same test in `main` to confirm the bug.
Then tried the solution and it works, so looks good to me.
I have left a tiny suggestion for a test.

Props for the clear instructions to reproduce + explanation how it works.

File: spec/models/tag_spec.rb (hunk 1 above)
Comment (Ridhwana) [reply spec/models/tag_spec.rb:232]: Great suggestion, my tunnel vision stopped me from realizing this, thank you will update accordingly!

**Review by dukegreene — APPROVED** [review]
Thanks for the Loom walkthrough and super clear QA setup instructions!

Works well locally and I can see the difference between `main`'s behavior and your branch's counts.

**Pull request merged**